precompile/contracts/ibc/client_test.go

```diff
@@ -0,0 +1,550 @@
+
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	commitmenttypes "github.com/cosmos/ibc-go/v7/modules/core/23-commitment/types"
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+	ibctm "github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+)
+
+func (suite *KeeperTestSuite) TestCreateClient() {
+	cases := []struct {
+		msg            string
+		clientState    exported.ClientState
+		consensusState exported.ConsensusState
+		expPass        bool
+		nextClientSeq  uint64
+	}{
+		{
+			"success: 07-tendermint client type supported",
+			ibctm.NewClientState(testChainID, ibctm.DefaultTrustLevel, trustingPeriod, ubdPeriod, maxClockDrift, testClientHeight, commitmenttypes.GetSDKSpecs(), ibctesting.UpgradePath),
+			suite.consensusState,
+			true,
+			3454,
+		},
+	}
+
+	statedb, _ := state.New(common.Hash{}, state.NewDatabase(rawdb.NewMemoryDatabase()), nil)
+	statedb.Finalise(true)
+	vmctx := vm.BlockContext{
+		CanTransfer: func(vm.StateDB, common.Address, *big.Int) bool { return true },
+		Transfer:    func(vm.StateDB, common.Address, common.Address, *big.Int) {},
+	}
+	vmenv := vm.NewEVM(vmctx, vm.TxContext{}, statedb, params.TestChainConfig, vm.Config{ExtraEips: []int{2200}})
+
+	for _, tc := range cases {
+		var input []byte
+
+		// clientStateLen     - clientState
+		clientState, ok := tc.clientState.(*ibctm.ClientState)
+		if !ok {
+			suite.Require().NoError(errors.New("convert to proto.Message failer"))
+		}
+		clientStateByte, err := clientState.Marshal()
+
+		suite.Require().NoError(err)
+
+		b := make([]byte, 8)
+		binary.BigEndian.PutUint64(b, tc.nextClientSeq)
+		vmenv.GetStateDB().SetPrecompileState(common.BytesToAddress([]byte("nextClientSeq")), b)
+
+		// 8 byte             - clientStateLen
+		clientStateLen := make([]byte, 8)
+		binary.BigEndian.PutUint64(clientStateLen, uint64(len(clientStateByte)))
+
+		input = append(input, clientStateLen...)
+		input = append(input, clientStateByte...)
+
+		// consensusStateLen  - consensusState
+
+		consensusState, ok := tc.consensusState.(*ibctm.ConsensusState)
+		if !ok {
+			suite.Require().NoError(errors.New("convert to proto.Message failer"))
+		}
+		consensusStateByte, err := consensusState.Marshal()
+		suite.Require().NoError(err)
+
+		// 8 byte             - consensusStateLen
+		consensusStateLen := make([]byte, 8)
+		binary.BigEndian.PutUint64(consensusStateLen, uint64(len(consensusStateByte)))
+
+		input = append(input, consensusStateLen...)
+		input = append(input, consensusStateByte...)
+
+		input = append(getCreateClientSignature, input...)
+		admin := allowlist.TestAdminAddr
+		enableds := allowlist.TestEnabledAddr
+
+		allowlist.SetAllowListRole(vmenv.StateDB, ContractAddress, admin, allowlist.AdminRole)
+
+		contract := createIbcGoPrecompile()
+
+		suppliedGas := uint64(10000000)
+		output, _, err := contract.Run(vmenv, admin, enableds, input, suppliedGas, false)
```

Comment:
Instead of all of the calls to BigEndian, I suggest making a separate function which can assemble the 'input' argument to the `contract.Run` function. That way it's easier to invoke.

Reply:
we are doing refactoring for this par, going to push new encoding/decoding logic soon